src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -156,30 +156,43 @@ export class UserService {
     return UserDtoMapper.mapRef(user, refCount, refCountActive);
   }
 
-  async createUser(
-    { address, signature, usedRef }: CreateUserDto,
-    userIp: string,
-    userOrigin?: string,
-    wallet?: Wallet,
-    specialCode?: string,
-  ): Promise<User> {
+  async createUser({
+    userDetails: { address, signature, usedRef },
+    userIp,
+    userOrigin,
+    wallet,
+    specialCode,
+    userData,
+  }: {
+    userDetails: CreateUserDto;
+    userIp: string;
+    userOrigin?: string;
+    wallet?: Wallet;
+    specialCode?: string;
+    userData?: UserData;
+  }): Promise<User> {
     let user = this.userRepo.create({ address, signature, addressType: CryptoService.getAddressType(address) });
+    const userIsActive = userData?.status === UserDataStatus.ACTIVE;
 
     user.ip = userIp;
     user.ipCountry = this.geoLocationService.getCountry(userIp);
     user.wallet = wallet ?? (await this.walletService.getDefault());
     user.usedRef = await this.checkRef(user, usedRef);
     user.origin = userOrigin;
+    user.status = userIsActive ? UserStatus.ACTIVE : UserStatus.NA;
 
```

Comment:
Maybe this is better (not setting the default)?
`userIsActive && (user.status = UserStatus.ACTIVE);`